third_party/terraform/resources/resource_container_cluster.go.erb

```diff
@@ -2113,22 +2123,38 @@ func cleanFailedContainerCluster(d *schema.ResourceData, meta interface{}) error
 	return nil
 }
 
-func waitForContainerClusterReady(config *Config, project, location, clusterName string, timeout time.Duration) error {
-	return resource.Retry(timeout, func() *resource.RetryError {
+var containerClusterRestingStates = RestingStates{
+	"RUNNING":  ReadyState,
+	"DEGRADED": ErrorState,
+	"ERROR":    ErrorState,
+}
+
+func containerClusterAwaitRestingState(config *Config, project, location, clusterName string, timeout time.Duration) (state string, stateType StateType, err error) {
+	err = resource.Retry(timeout, func() *resource.RetryError {
 		name := containerClusterFullName(project, location, clusterName)
 		cluster, err := config.clientContainerBeta.Projects.Locations.Clusters.Get(name).Do()
```

Comment:
nit: local variables with the same name as named return variables is a warning flag for me. The scoping looks safer here but I'd suggest renaming 1 of them for clarity.